src/CredentialRepository.php

```diff
@@ -154,7 +154,9 @@ public function hasChanged(): bool
     protected function setCredentials(array $credentials): void
     {
         $this->credentials = array_map(function ($data) {
-            $data['source'] = PublicKeyCredentialSource::createFromArray($data['source']);
+            if (is_array($data['source'])) {
+                $data['source'] = PublicKeyCredentialSource::createFromArray($data['source']);
```

Comment:
What's the scenario where `$data['source']` is not an array? Don't we need to do something when that happens?

Reply:
I take it that `$data['source']` will already be a `PublicKeyCredentialCreationOptions` in that scenario?

Reply:
> What's the scenario where $data['source'] is not an array?
I *think* it's caused by this 
```php
    public function __serialize(): array
    {
        return [
            'memberID' => $this->memberID,
            'credentials' => $this->toArray() # <<<<<
        ];
    }
```
https://github.com/silverstripe/silverstripe-webauthn-authenticator/blob/4/src/CredentialRepository.php#L202